src/Development/IDE/Spans/Documentation.hs

```diff
@@ -31,13 +38,32 @@ getDocumentationTryGhc
 #if MIN_GHC_API_VERSION(8,6,0) && !defined(GHC_LIB)
 getDocumentationTryGhc sources name = do
   res <- catchSrcErrors "docs" $ getDocs name
+  uris <- getUris
   case res of
-    Right (Right (Just docs, _)) -> return $ SpanDocString docs
-    _ -> return $ SpanDocText $ getDocumentation sources name
+    Right (Right (Just docs, _)) -> return $ (SpanDocString docs uris)
+    _ -> return $ (SpanDocText (getDocumentation sources name) uris)
 #else
 getDocumentationTryGhc sources name = do
-  return $ SpanDocText $ getDocumentation sources name
+  uris <- getUris
+  return $ SpanDocText (getDocumentation sources name) uris
 #endif
+  where
+    -- Get the uris to the documentation and source html pages if they exist
+    getUris = do
+      df <- getSessionDynFlags
+      (docFp, srcFp) <-
+        case nameModule_maybe name of
+          Just mod -> liftIO $ do
```

Comment:
This requires calling into `IO`, which means now that after https://github.com/bubba/ghcide/commit/7dc6e2678a58d4822903fda8172329dd252b3629 I'm not sure how to integrate this with doc local completions, which seems to be a function purely based off of the `ParsedModule` cc @pepeiborra

Reply:
Local completions can use `IO` so I presume that's not what is blocking you.
 
Local completions do not use the ghc api to extract documentation, i.e. they do not call `getDocumentationTryGhc`, since that requires a `ModIface` which is only available after type checking. Instead, they get the documentation from the parse tree by calling `D.IDE.Spans.Documentation.getDocumentation`. 

So I think the problem is that there are two code paths to get documentation and this PR needs to be extended to cover both.

Reply:
It also needs to get access to the `DynFlags` which I'm not sure how to get outside of the GHC session, i.e. in `getDocumentation`. In `localCompletionsForParsedModule`

Reply:
Do we want to run this code for local completions anyway? Locally defined identifiers will not have any documentation links

Reply:
Yeah I was thinking that too. The PR currently just ignores it in `Logic.hs`:

```haskell
    mkComp n ctyp ty =
         CI ctyp pn thisModName ty pn Nothing doc (ctyp `elem` [CiStruct, CiClass])
       where
         pn = ppr n
         doc = SpanDocText $ getDocumentation [pm] n
         doc = SpanDocText (getDocumentation [pm] n) (SpanDocUris Nothing Nothing)
```